extensions/positron-python/python_files/positron/positron_ipykernel/patch/bokeh.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+# Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+#
+
+from logging import Logger
+
+from ..session_mode import SessionMode
+
+MIME_TYPE_POSITRON_WEBVIEW_FLAG = "application/positron-webview-load.v0+json"
+
+
+def handle_bokeh_output(session_mode: SessionMode, logger: Logger) -> None:
+    """
+    Override the bokeh notebook display function to add a flag that the front-end can pick up on to
+    know that the data coming over should be replayed in multiple steps.
+
+    Args:
+        session_mode (SessionMode): The mode that the kernel application was started in.
+        logger (Callable[[str], None]): A logger function.
+    """
+    if session_mode == SessionMode.NOTEBOOK:
+        # Don't do anything if we're in a notebook
+        return
+
+    try:
+        from bokeh.io import notebook
+    except ImportError:
+        pass
```

Comment:
Super duper nit: I think this could `return`, then you don't need the `else` and below can be dedented.